synapse/handlers/register.py

```diff
@@ -386,11 +386,32 @@ async def _create_and_join_rooms(self, user_id: str) -> None:
                 room_alias = RoomAlias.from_string(r)
 
                 if self.hs.hostname != room_alias.domain:
+                    # If the alias is remote, try to join the room. This might fail
+                    # because the room might be invite only, but we don't have any local
+                    # user in the room to invite this one with, so at this point that's
+                    # the best we can do.
                     logger.warning(
```

Comment:
```suggestion
                    logger.info(
```

Since it's no longer a failure mode.

Reply:
It is still a misconfiguration, so imho something we should warn the user about.

Reply:
Oh, is it not a feature that you can auto-join remote rooms?

Reply:
I'm a bit conflicted about this, as this should only happen if you've set "auto create rooms" on *and* specified a remote alias, which obviously won't work. However, its not totally unreasonable to configure a local and remote room to auto join, and you want to auto create the local one.

Reply:
> Oh, is it not a feature that you can auto-join remote rooms?

It is, but my point was that this code is for creating and joining the room, not just joining them (hence the warning about _creating_ a room).

> However, its not totally unreasonable to configure a local and remote room to auto join, and you want to auto create the local one.

That's a fair point. I'll downgrade to info.